assets/src/editor/events/dates-and-tickets-metabox/dates-and-tickets-metabox.js

```diff
@@ -91,13 +92,11 @@ const MetaBox = ( { eventId } ) => {
 	);
 };
 
-const DatesAndTicketsMetabox = withEditorInitialization( MetaBox );
-
-MetaBox.propTypes = {
+DatesAndTicketsMetabox.propTypes = {
```

Comment:
Hmm, I think we should have the propTypes defined for the composed component (which I know wasn't the case originally either).

Reply:
what are the benefits and/or what is the logic behind doing that?
Those props are part of of the component, not the HOC, so logically (to me) it doesn't make sense to associate them with the HOC.

As well, from a troubleshooting point of view, if a notice appeared in the editor saying there was a bad prop type for `withEditorInitialization` one might open up `/assets/src/editor/events/initialization/index.js` instead of this file, which would potentially just cause confusion and waste time by forcing a search for the real source of the problem. Leaving it as is will indicate an issue with the props for **this** component and **this** file.

Reply:
I'm not saying you add the prop types on to `withEditorInitialization`.  I'm saying you add the PropTypes to the **composed** component (`withEditorInitialization( DatesAndTicketsMetabox )` ).  The composed component is what gets exported for the public api, so imo that's what should have the propTypes. Does that clarify?

Reply:
I realize that, although my comment about a console warning appearing for `withEditorInitialization` does look like I thought you meant to define the prop types there. I was assuming that a call stack for any warning would indicate that the error occurred when attempting to render `withEditorInitialization` even though it wasn't happening in the file that contains that HOC. I could be completely wrong about that and any notices would be 100% clear that the issue was in `dates-and-tickets-metabox.js` but I assumed that was not the case and that the potential for confusion exists.

So that said, I still don't see the logic or any benefit to defining the prop types for the composed component.

> re: The composed component is what gets exported for the public api,

sure that's currently the case, but what if that changes? What if another HOC gets added and we switch to using compose?
Similar to my comment below, what I'm doing here is in line  with the way we have been doing things in every other file, so why the change here? If there is a good reason to make this change then are we going to do so in every other composed component?

Reply:
- The public api exposes `withEditorInitialization( DatesAndTicketsMetabox )`.  So usage will be `<ComposedComponent eventId={ 10 } />`.  I think any errors should be on what is publicly exposed.

> but what if that changes? What if another HOC gets added and we switch to using compose?

Then any additional prop type would still be defined on the composed component.

> Similar to my comment below, what I'm doing here is in line with the way we have been doing things in every other file, so why the change here? If there is a good reason to make this change then are we going to do so in every other composed component?

While consideration _should_ be given for what has already been done, I don't think that precludes noticing things in reviews and improving what's being done.  I noticed this, commented on it and made a suggestion for improving it.  We definitely don't _have_ to go through and make the changes everywhere else (which I don't think are as big a deal as you might think... proptypes themselves have been used inconsistently so far), but I do think its worthwhile to make any appropriate changes as they are spotted.

With that said, I'll reiterate what I said in my approval comment:

>  I'll leave it up to you whether you action on the comments.